src/containers/Reports/Reports.js

```diff
@@ -1,125 +1,191 @@
-import React, { Component } from 'react';
-import { connect } from 'react-redux';
+import React, { Component } from "react";
+import { connect } from "react-redux";
 import {
-  LineChart, Line, XAxis, YAxis, CartesianGrid, Tooltip, Legend, ResponsiveContainer
-} from 'recharts';
-import { Row, Col, Modal, Icon, Statistic, Typography } from 'antd';
-import axios from '../../_util/axios-api';
-import { authHeader } from '../../_util/auth/auth-header';
-import { displayName } from '../../_util/district';
-
+  LineChart,
+  Line,
+  XAxis,
+  YAxis,
+  CartesianGrid,
+  Tooltip,
+  Legend,
+  ResponsiveContainer,
+} from "recharts";
+import { Row, Col, Icon, Statistic, Typography } from "antd";
+import axios from "../../_util/axios-api";
+import { authHeader } from "../../_util/auth/auth-header";
+import { displayName, getAssociatedSenators } from "../../_util/district";
 
 class Reports extends Component {
+  state = {
+    statistics: null,
+  };
 
-    state = {
-        statistics: null
+  componentDidMount() {
+    if (this.state.statistics === null) {
+      this.fetchStatistics();
     }
+  }
 
-    componentDidMount() {
-        if (this.state.statistics === null) {
-            this.fetchStatistics();
-        }
-    }
-
-    fetchStatistics(cb) {
-        if (this.props.district) {
-            const requestOptions = {
-                url: `/stats/${this.props.district.districtId}`,
-                method: 'GET',
-                headers: { ...authHeader(), 'Content-Type': 'application/json' },
-            };
-            axios(requestOptions).then((response)=>{
-                const statistics = response.data;
-                this.setState({statistics: statistics});
-            }).catch((e) => {
-                Modal.error({
-                    title: "Error Loading Page",
-                    content: e.message,
-                });
-                this.setState({fetchError: e.message})
+  fetchStatistics() {
+    if (this.props.districts) {
+      const districts = this.props.districts;
+      let promises = [];
+      let responses = [];
+      for (let index = 0; index < districts.length; ++index) {
+        if (districts[index]) {
+          const requestOptions = {
+            url: `/stats/${districts[index].districtId}`,
+            method: "GET",
+            headers: { ...authHeader(), "Content-Type": "application/json" },
+          };
+          promises.push(
+            axios(requestOptions).then((response) => {
+              responses.push(response.data);
             })
+          );
         }
+      }
+      Promise.all(promises).then(() => {
+        this.setState({ statistics: responses });
+      });
```

Comment:
The error handling has been removed. The page needs to show an error message if the data is not able to be retrieved.